Sources/SwiftProtobuf/Message+StreamDecodingIterator.swift

```diff
@@ -0,0 +1,131 @@
+// Sources/SwiftProtobuf/Message+StreamDecodingIterator.swift - Iterator over binary delimited input streams
+//
+// Copyright (c) 2023 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/main/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// Extensions to `Message` to provide binary coding and decoding using ``Foundation/Data``.
+///
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+extension Message {
+  
+  public static func streamDecodingIterator(inputStream: InputStream, bufferLength: Int = 32768) -> StreamDecodingIterator<Self> {
+    return StreamDecodingIterator<Self>(inputStream: inputStream, bufferLength: bufferLength)
+  }
+}
+
+/**
+ Iterates over a binary-delimited protobuf input stream
+ */
+public struct StreamDecodingIterator<M: Message>: IteratorProtocol {
+  
+  fileprivate let buffer: ResizingBuffer
+  
+  public init(inputStream: InputStream, bufferLength: Int = 32768) {
+    buffer = ResizingBuffer(inputStream: inputStream, bufferLength: bufferLength)
+  }
+  
+  public func next() -> M? {
+    do {
+      let messageLength = try decodeVarint()
+      if (messageLength == 0) {
+        return nil
+      }
+      var message = M()
+      let data = try buffer.read(for: Int(messageLength))
+      try message.merge(serializedData: data)
+      return message
+    } catch {
+      print(error)
```

Comment:
Probably don't want to `print` here. I guess there's no way to expose to the iterator when the stream was corrupted vs. just ended?

Reply:
Yes I wasn't sure what to do here as ```IteratorProtocol.next``` can't throw. Maybe setting a status field on the iterator would do the trick?

Reply:
To deal with error-handling I added an optional StreamErrorDelegate to the iterator initializer - demonstrated in the test suite.

Reply:
That seems like a reasonable approach.